lib/graphiti/rails.rb

```diff
@@ -19,77 +20,21 @@ def self.included(_klass)
-    autoload :Debugging, "graphiti/rails/debugging"
-    autoload :ExceptionsApp, "graphiti/rails/exceptions_app"
-    autoload :DefaultExceptionHandler, "graphiti/rails/exception_handlers"
-    autoload :InvalidRequestExceptionHandler, "graphiti/rails/exception_handlers"
-
-    # The default exception handler for Graphiti::Rails errors.
-    # In general, you should not need to call or modify this.
-    # @private
-    def self.default_exception_handler
-      DefaultExceptionHandler
-    end
-
-    if defined?(Graphiti::Errors::InvalidRequest)
-      register_exception Graphiti::Errors::InvalidRequest,
-        handler: InvalidRequestExceptionHandler
-    end
-
-    # Returns a boolean of whether the specified content type should be handled by a Graphiti exception handler.
-    # The list of format can be found in `.handled_exception_formats`.
-    # @param content_type [Mime::Type]
-    # @return [Boolean]
-    def self.render_exception_for_format?(content_type)
-      handled_exception_formats.include?(content_type.to_sym)
-    end
-
-    # @return [GraphitiErrors::ExceptionHandler]
-    def self.exception_handler_for(exception)
-      # TODO: Move this into GraphitiErrors
-      _errorable_registry[exception.class] || default_exception_handler.new
-    end
-
-    # @param [Exception] exception
-    # @param [Boolean] show_raw_error (false)
-    # @return [Array<Integer, Hash>] HTTP status and payload
-    def self.exception_details(exception, show_raw_error: false)
-      # TODO: Move this into GraphitiErrors
-      exception_klass = exception_handler_for(exception)
-      exception_klass.show_raw_error = show_raw_error
-      status = exception_klass.status_code(exception)
-      payload = exception_klass.error_payload(exception)
-      [status, payload]
-    end
-
-    # @param [Exception] exception
-    # @param [Mime::Type] content_type
-    # @param [Hash] keywords passed through to {.exception_details}
-    # @return [Array<Integer, Mime::Type, String>] HTTP status, content type, and payload
-    def self.rendered_exception(exception, content_type:, **keywords)
-      status, body = exception_details(exception, **keywords)
-
-      to_format = (content_type == :jsonapi) ? "to_json" : "to_#{content_type.to_sym}"
-
-      if content_type && body.respond_to?(to_format)
-        formatted_body = body.public_send(to_format)
-        format = content_type
-      else
-        formatted_body = body.to_json
-        format = Mime[:json]
-      end
-
-      [status, format, formatted_body]
-    end
   end
 end
 
 ActiveSupport.on_load(:action_controller) do
   include Graphiti::Rails::Context
   include Graphiti::Rails::Debugging
+
+  # A global handler here is somewhat risky. However, we default to using Rails-style rendering so it shouldn't cause
+  # changes to what people are expecting. We should write more tests though.
+  register_exception Exception, status: :passthrough, handler: Graphiti::Rails::ExceptionHandler
```

Comment:
I don't love doing this, but I'm thinking it's probably ok?

Reply:
Ok, there's actually an issue here in that the default behavior would be to pass on to `config.exceptions_app` which might have different behavior, so this needs more thought.

Reply:
See the newest approach.